app/uk/gov/hmrc/cgtpropertydisposalsfrontend/controllers/returns/yeartodatelliability/YearToDateLiabilityController.scala

```diff
@@ -0,0 +1,1110 @@
+import uk.gov.hmrc.cgtpropertydisposalsfrontend.models.returns.YearToDateLiabilityAnswers.NonCalculatedYearToDateLiabilityAnswers.{CompleteNonCalculatedYearToDateLiabilityAnswers, IncompleteNonCalculatedYearToDateLiabilityAnswers}
+import uk.gov.hmrc.cgtpropertydisposalsfrontend.models.returns.YearToDateLiabilityAnswers.{CalculatedYearToDateLiabilityAnswers, NonCalculatedYearToDateLiabilityAnswers}
+import uk.gov.hmrc.cgtpropertydisposalsfrontend.models.returns._
+import uk.gov.hmrc.cgtpropertydisposalsfrontend.models.{BooleanFormatter, ConditionalRadioUtils, FormUtils, SessionData}
+import uk.gov.hmrc.cgtpropertydisposalsfrontend.repos.SessionStore
+import uk.gov.hmrc.cgtpropertydisposalsfrontend.services.returns.{CgtCalculationService, ReturnsService}
+import uk.gov.hmrc.cgtpropertydisposalsfrontend.util.Logging.LoggerOps
+import uk.gov.hmrc.cgtpropertydisposalsfrontend.util.{Logging, toFuture}
+import uk.gov.hmrc.cgtpropertydisposalsfrontend.views.html.returns.{ytdliability => pages}
+import uk.gov.hmrc.play.bootstrap.controller.FrontendController
+
+import scala.concurrent.{ExecutionContext, Future}
+
+@Singleton
+class YearToDateLiabilityController @Inject() (
+  val authenticatedAction: AuthenticatedAction,
+  val sessionDataAction: SessionDataAction,
+  val sessionStore: SessionStore,
+  val errorHandler: ErrorHandler,
+  returnsService: ReturnsService,
+  cgtCalculationService: CgtCalculationService,
+  cc: MessagesControllerComponents,
+  val config: Configuration,
+  estimatedIncomePage: pages.estimated_income,
+  personalAllowancePage: pages.personal_allowance,
+  hasEstimatedDetailsPage: pages.has_estimated_details,
+  taxDuePage: pages.tax_due,
+  mandatoryEvidencePage: pages.upload_mandatory_evidence,
+  calculatedCheckYouAnswersPage: pages.calculated_check_your_answers,
+  taxableGainOrLossPage: pages.taxable_gain_or_loss,
+  nonCalculatedCheckYourAnswersPage: pages.non_calculated_check_your_answers
+)(implicit viewConfig: ViewConfig, ec: ExecutionContext)
+    extends FrontendController(cc)
+    with WithAuthAndSessionDataAction
+    with Logging
+    with SessionUpdates {
+
+  private def withFillingOutReturnAndYTDLiabilityAnswers(
+    request: RequestWithSessionData[_]
+  )(
+    f: (
+      SessionData,
+      FillingOutReturn,
+      YearToDateLiabilityAnswers
+    ) => Future[Result]
+  ): Future[Result] =
+    request.sessionData.flatMap(s => s.journeyStatus.map(s -> _)) match {
+      case Some((s, r @ FillingOutReturn(_, _, _, d: SingleDisposalDraftReturn))) =>
+        d.yearToDateLiabilityAnswers match {
+          case Some(y) => f(s, r, y)
+
+          case None =>
+            d.reliefDetailsAnswers match {
+              case Some(CompleteReliefDetailsAnswers(_, _, Some(_: OtherReliefsOption.OtherReliefs))) =>
```

Comment:
this is new - here we're deciding if the user should be calculated or non-calculated in a single disposal journey